src/backend/azure.rs

```diff
@@ -70,6 +77,61 @@ const AZURE_BLOB_TYPE: &str = "BlockBlob";
 /// The name of the root container.
 const AZURE_ROOT_CONTAINER: &str = "$root";
 
+/// Inserts the authentication header to the request.
+fn insert_authentication_header(auth: &AzureAuthConfig, request: &mut Request) -> Result<()> {
+    let signer = RequestSigner::new(auth);
+    let auth = signer.sign(request).ok_or(AzureError::InvalidAccessKey)?;
+    request.headers_mut().insert(
+        header::AUTHORIZATION,
+        HeaderValue::try_from(auth).expect("value should be valid"),
```

Comment:
Since this returns a `Result` already, should we instead propagate that error?

Reply:
In this case I think an expectation is fine; we should never generate an `Authorization` header value that isn't valid (it's in no way a user error).

Reply:
Alright, in honesty, my skimming of the docs around the failure conditions here left me with an incomplete understanding of what would make it invalid. But if it's not something you expect to run into, this seems fine.